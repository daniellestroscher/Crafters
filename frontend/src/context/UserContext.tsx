import { createContext } from 'react';
import { user } from '../types/User';

export const defaultUser = {
  auction_wins: 0,
  bio: '',
  createdAt: '',
  email: '',
  id: 0,
  name: '',
  public_picture_id: '',
  updatedAt: '',
  userPicUrl: '',
  username: '',
};

export const UserContext = createContext<{ userData: user; setUserData: (arg: user) => void }>({
  userData: defaultUser,
  setUserData: () => {},
});
