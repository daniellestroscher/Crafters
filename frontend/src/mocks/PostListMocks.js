export const mockPostList = [{
  createdAt: "2022-11-18T16:00:14.922Z",
  description: "i love palm trees",
  id: 1,
  postPicUrl: "https://res.cloudinary.com/dovw0n8pd/image/upload/v1668787214/knittingApp/posts/aftczo6yh0zuqednxmpk.jpg",
  price: "1000",
  public_image_id: "knittingApp/posts/aftczo6yh0zuqednxmpk",
  sold: false,
  title: "a palm tree",
  updatedAt: "2022-11-18T16:00:14.922Z",
  userEmail: "d@test.com",
  username: 'Tony'
}, {
  createdAt: "2022-11-18T16:22:20.047Z",
  description: "I also love coffee in Santa Cruz",
  id: 2,
  postPicUrl: "https://res.cloudinary.com/dovw0n8pd/image/upload/v1668788539/knittingApp/posts/aektblsljswdavahgafj.jpg",
  price: "100000",
  public_image_id: "knittingApp/posts/aektblsljswdavahgafj",
  sold: false,
  title: "Verve",
  updatedAt: "2022-11-18T16:22:20.047Z",
  userEmail: "d@test.com",
  username: 'Danny'
}]

