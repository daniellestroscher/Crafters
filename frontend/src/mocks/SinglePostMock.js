export const singleMock = {
  createdAt: "2022-11-18T16:22:20.047Z",
  description:"I also love coffee in Santa Cruz",
  id:2,
  postPicUrl:"https://res.cloudinary.com/dovw0n8pd/image/upload/v1668788539/knittingApp/posts/aektblsljswdavahgafj.jpg",
  price:"100000",
  public_image_id:"knittingApp/posts/aektblsljswdavahgafj",
  sold:false,
  title:"Verve",
  updatedAt:"2022-11-18T16:22:20.047Z",
  userEmail:"d@test.com"
}

export const singleMockNew = {
  createdAt: "2022-11-18T16:22:20.047Z",
  description:"YAYAYA",
  id:99,
  postPicUrl:"https://res.cloudinary.com/dovw0n8pd/image/upload/v1668788539/knittingApp/posts/aektblsljswdavahgafj.jpg",
  price:"5500",
  public_image_id:"knittingApp/posts/aektblsljswdavahgafj",
  sold:false,
  title:"a beautiful post",
  updatedAt:"2022-11-18T16:22:20.047Z",
  userEmail:"d@test.com"

}