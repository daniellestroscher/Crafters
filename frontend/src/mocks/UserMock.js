export const userMock = {
  email:"d@test.com",
  email_verified:false,
  name:"d@test.com",
  nickname:"d",
  picture:"https://s.gravatar.com/avatar/0fb1d94c049e854fd15b9cd629bac718?s=480&r=pg&d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fd.png",
  sub:"auth0|6376cdf467ad3e09030c8e0d",
  updated_at:"2022-11-18T00:12:36.781Z",
}

export const contextUserMock = {
  auction_wins:0,
  bio:"",
  createdAt:"2022-11-18T00:35:04.169Z",
  email:"d@test.com",
  id:1,
  name:"danielle",
  public_picture_id:"knittingApp/users/kmyeziqldhrczp0vzi75",
  updatedAt:"2022-11-18T16:15:10.759Z",
  userPicUrl:"https://res.cloudinary.com/dovw0n8pd/image/upload/v1668788110/knittingApp/users/kmyeziqldhrczp0vzi75.jpg",
  username:"danielle",
}