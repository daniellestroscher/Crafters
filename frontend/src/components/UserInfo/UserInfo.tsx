import { FC, useContext } from 'react';
import { Link } from 'react-router-dom';

import './UserInfo.css';

import { UserContext } from '../../context/UserContext';

type UserInfoProps = {
  user: {
    email?: string,
    email_verified?: boolean,
    name: string,
    nickname: string,
    picture: string,
    sub?: string,
    updated_at?: string,
  };
  postNumber: number;
  numberOfFavs: number;
}

export const UserInfo: FC<UserInfoProps> = ({ user, postNumber, numberOfFavs }) => {
  const { userData } = useContext(UserContext);
  console.log(userContext);

  return (
    <div className="singleFlexProp">
      <div className="profileInfo__image">
        <img
          loading="lazy"
          src={userData && userData.userPicUrl ? userData.userPicUrl : user.picture}
          alt={userData && userData.username ? userData.username : user.nickname}
        />
      </div>

      <div className="profileInfo__user">
        <div className="userInfo__container">
          <h2 className="userInfo__username">{userData && userData.username ? userData.username : user.nickname}</h2>
          <Link to="/update-profile" className="editButtonInfo" data-testid="edit-profile-link">
            Edit profile
          </Link>
        </div>
        <div className="statsInfo">
          <p className="statistics">
            <span className="individualStats">{postNumber}</span> posts
          </p>
          <p className="statistics">
            <span className="individualStats">{numberOfFavs} </span> favs
          </p>
        </div>
        <p className="user__fullName">{userData && userData.name ? userData.name : user.name}</p>
      </div>
    </div>
  );
};
