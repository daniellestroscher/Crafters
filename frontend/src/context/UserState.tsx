import React, { useState, useEffect, FC, PropsWithChildren, ReactNode } from 'react';
import { useAuth0 } from '@auth0/auth0-react';

import { UserContext } from './UserContext';
import { user } from '../types/User';
import { defaultUser } from './UserContext';
import { retrieveUser, storeUser } from '../services/fetchData';

interface Props {
  children: ReactNode;
}
export const UserState: FC<Props> = ({ children }) => {
  const { isAuthenticated, user } = useAuth0();
  const [userData, setUserData] = useState<user>(defaultUser);

  useEffect(() => {
    if (isAuthenticated && user) {
      retrieveUser(user.email).then(data => {
        if (data) {
          setUserData(data);
        } else {
          const newUser = {
            id: 0,
            email: user.email,
            bio: '',
            username: user.nickname,
            name: '',
            userPicUrl: user.picture,
          };

          storeUser(newUser).then(info => {
            if (!info.error) {
              setUserData(info);
            }
          });
        }
      });
    }
  }, [isAuthenticated, user]);

  return <UserContext.Provider value={{ userData, setUserData }}>{children}</UserContext.Provider>;
};
