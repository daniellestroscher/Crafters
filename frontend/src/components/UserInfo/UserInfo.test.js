import { render } from "@testing-library/react";
import { UserInfo } from "./UserInfo";
import { expect } from '@jest/globals';
import { screen } from '@testing-library/react';
import { UserContext } from '../../context/UserContext';
import { userMock, contextUserMock } from '../../mocks/UserMock';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

describe('UserInfo component', () => {
  test('it should render the component with an Edit profile Button', () => {
    const userData = contextUserMock;
    const user = userMock;
    const postNumber = 5;
    const numberOfFavs = 5;
    render(
      <UserContext.Provider value={{ userData }}>
        <UserInfo user={user} postNumber={postNumber} numberOfFavs={numberOfFavs}/>
      </UserContext.Provider>, {wrapper: BrowserRouter}
    )
    expect(screen.getByText('Edit profile')).toBeInTheDocument();
    expect(screen.getByText('favs')).toBeInTheDocument();
    expect(screen.getByText('posts')).toBeInTheDocument();
  })

  test('it should change pages on button click', async () => {
    const userData = contextUserMock;
    const user = userMock;
    const postNumber = 5;
    const numberOfFavs = 5;

    const component =  renderer.create(
      <MemoryRouter>
        <UserContext.Provider value={{ userData }}>
          <UserInfo user={user} postNumber={postNumber} numberOfFavs={numberOfFavs}/>
        </UserContext.Provider>
      </MemoryRouter>
    ).toJSON();

    expect(component).toMatchSnapshot();

  })
});