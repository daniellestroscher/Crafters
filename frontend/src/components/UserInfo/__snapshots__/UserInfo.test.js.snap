// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component it should change pages on button click 1`] = `
<div
  className="singleFlexProp"
>
  <div
    className="profileInfo__image"
  >
    <img
      alt="danielle"
      loading="lazy"
      src="https://res.cloudinary.com/dovw0n8pd/image/upload/v1668788110/knittingApp/users/kmyeziqldhrczp0vzi75.jpg"
    />
  </div>
  <div
    className="profileInfo__user"
  >
    <div
      className="userInfo__container"
    >
      <h2
        className="userInfo__username"
      >
        danielle
      </h2>
      <a
        className="editButtonInfo"
        data-testid="edit-profile-link"
        href="/update-profile"
        onClick={[Function]}
      >
        Edit profile
      </a>
    </div>
    <div
      className="statsInfo"
    >
      <p
        className="statistics"
      >
        <span
          className="individualStats"
        >
          5
        </span>
         posts
      </p>
      <p
        className="statistics"
      >
        <span
          className="individualStats"
        >
          5
           
        </span>
         favs
      </p>
    </div>
    <p
      className="user__fullName"
    >
      danielle
    </p>
  </div>
</div>
`;
